### ISSUE TITLE
SSLProxyPass?

### ISSUE THREAD

**Issue by muhahacz** [webmin/webmin#256]
Hi, i am using latest webmin with authentic theme 16.00.
I am trying to setup ssl proxy pass, but it seems, that every post/cgi request, for example (https://10.0.0.47:9090/webmin/sshd/apply.cgi?), or all requests from save buttons, ends in cancelled state.

Chrome says:
Mixed Content: The page at 'https://10.0.0.47:9090/webmin/' was loaded over HTTPS, but requested an insecure form action 'http://10.0.0.47:9090/sshd/'. This request has been blocked; the content must be served over HTTPS.

 Any advice how to handle these requests from ssl ? 

 Thanks
 

```
<VirtualHost *:9090>
    ServerAdmin admin@aexample.io
    ServerName aexample.io
    ErrorLog /var/log/2testproxy_error_log
    CustomLog /var/log/2testproxy_log common

    SSLEngine On
    SSLProxyEngine On

    SSLCertificateFile /etc/apache2/ssl/apache.crt
    SSLCertificateKeyFile /etc/apache2/ssl/apache.key

        <proxy *>
    Order Allow,Deny
    Allow from all
        </proxy>

        ProxyRequests off
        ProxyPreserveHost On

ProxyPass /webmin/ http://127.0.0.1:10000/
ProxyPassReverse /webmin/ http://127.0.0.1:10000/
</VirtualHost>
```

```
port=10000
root=/usr/share/webmin
mimetypes=/usr/share/webmin/mime.types
addtype_cgi=internal/cgi
realm=Webmin Server
logfile=/var/webmin/miniserv.log
errorlog=/var/webmin/miniserv.error
pidfile=/var/webmin/miniserv.pid
logtime=168
ppath=
ssl=0
no_ssl2=1
no_ssl3=1
env_WEBMIN_CONFIG=/etc/webmin
env_WEBMIN_VAR=/var/webmin
atboot=1
logout=/etc/webmin/logout-flag
listen=10000
denyfile=\.pl$
log=1
blockhost_failures=5
blockhost_time=60
syslog=1
session=1
premodules=WebminCore
server=MiniServ/1.760
userfile=/etc/webmin/miniserv.users
keyfile=/etc/apache2/ssl/apache.pem
passwd_file=/etc/shadow
passwd_uindex=0
passwd_pindex=1
passwd_cindex=2
passwd_mindex=4
passwd_mode=0
preroot=authentic-theme
passdelay=1
cipher_list_def=1
sudo=1
cookiepath=/webmin
logouttimes=
```
```
passwd_mindex=4
find_pid_command=ps auwwwx | grep NAME | grep -v grep | awk '{ print $2 }'
tempdelete_days=7
passwd_file=/etc/shadow
path=/bin:/usr/bin:/sbin:/usr/sbin:/usr/local/bin
passwd_uindex=0
passwd_pindex=1
passwd_cindex=2
by_view=0
ld_env=LD_LIBRARY_PATH
os_type=debian-linux
os_version=3.1
real_os_type=Ubuntu Linux
real_os_version=15.04
lang=en.UTF-8
log=1
referers_none=1
md5pass=1
theme=authentic-theme
product=webmin
webprefixnoredir=1
webprefix=/webmin
referer=127.0.0.1
```

**Comment by qooob** [comment]
How does it work for you in default theme? Do you get same error message?

**Comment by muhahacz** [comment]
Maybe this is a different error, but i tried it again on fresh install, latest webmin, ubuntu server 15.04 

I changed only these settings:

```
/etc/webmin/config
webprefixnoredir=1 
webprefix=/webmin
referer=127.0.0.1
```

```
/etc/webmin/miniserv.conf
cookiepath=/webmin
ssl=0
```
and even if i access to webmin miniserverer url with edited settings ( http://192.168.56.50:10000/webmin ) without proxy, page appear without css style.There are no 404 for css, picture attached

http://s30.postimg.org/oa6vx5y75/snapshot2.png

**Comment by jcameron** [comment]
Did you try changing to the older "Gray Framed Theme" in Webmin?

**Comment by muhahacz** [comment]
Even login does not work propetly, im not able to login, no matter of theme.
When i change name of gray-theme folder, old theme shows up, picture attached
http://s16.postimg.org/5korv4xic/image.jpg
I think it is easy reproducible, you can try.

I am wondered ,  why is  authentic-theme in /usr/share/webmin ? Even in this master branch is not authentic-theme available

**Comment by muhahacz** [comment]
Yes, after renaming  gray-theme folder, old theme shows up. Still, i can not login on: http://192.168.56.50:10000/webmin

Login (https://192.168.56.50:9090/webmin/) works fine on proxy, until i relogin, which will redirect me to 
http://192.168.56.50:10000

Clearly http://www.webmin.com/apache.html tutorial is not enough to setup proper ssl proxy pass

**Comment by muhahacz** [comment]
Look at these images, with this settings i have still some 404. It looks like subdir prefix is not fully supported
```
/etc/webmin/config
webprefixnoredir=1 
webprefix=/webmin
referer=127.0.0.1

/etc/webmin/miniserv.conf
cookiepath=/webmin
ssl=0
```
```
http://192.168.56.50:10000/webmin/
```
http://i.imgur.com/7r5dBtz.png
```
http://192.168.56.50:10000/
```
http://i.imgur.com/NlvNeEv.png

For example, there is request on 
http://192.168.56.50:10000/webmin/unauthenticated/sorttable.js
but in fact, sorttable.js is available on
http://192.168.56.50:10000/unauthenticated/sorttable.js

**Comment by muhahacz** [comment]
Seems that situation on older version of webmin 1.750 is different. 
Still, can you explain to me, why is on /webmin prefix left sidebar missing, or there is any dashboard ?
http://i.imgur.com/osZc52L.png
http://i.imgur.com/G31m21J.png

**Comment by muhahacz** [comment]
@jcameron  will you look at this bug, please?

**Comment by jcameron** [comment]
If you go direct to a URL like /webmin , then the left menu won't appear. Normally the /webmin path is served from the right frame, and the left frame contains a separate menu.

**Comment by muhahacz** [comment]
Thats one question, but will you look at bug described above? 
There are some 404s for static files when you using /webmin context in latest webmin. 1.750 is OK

**Comment by muhahacz** [comment]
@jcameron ? 

/var/webmin/miniserv.error attached
http://pastebin.com/Wi66jdaM

**Comment by jcameron** [comment]
So based on that log, the issue is that the /webmin path (which is presumably used to access your Webmin via Apache) is being passed through to Webmin itself, which is wrong.

I see from your original post that you're running an Apache virtualhost on port 9090. Other than proxying to Webmin, is that virtualhost used to serve any other content?

**Comment by muhahacz** [comment]
@jcameron Forget any posts above, any proxypass, 

It happens immediately / in latest version / when you set:

```
Add the lines webprefix=/webmin and webprefixnoredir=1 to /etc/webmin/config.
Add the line cookiepath=/webmin to /etc/webmin/miniserv.conf
```

Seems there is a bug, because v1.750 is OK with this settings.

**Comment by jcameron** [comment]
Ok, but why would you want to add those lines unless you are running Webmin behind an Apache proxy in a sub-directory? They only have validity in that configuration. That's why the specific ProxyPass directives are important.

**Comment by Storycrafter** [comment]
@muhahacz You're not the only one affected by this.  I don't care to run Apache as a proxy, but am using Nginx instead.  I am encountering the same behavior and issue as you.
